base_layer/core/src/mempool/service/service.rs

```diff
@@ -173,13 +198,36 @@ impl MempoolService {
             source_peer_id,
         );
         let mut inbound_handlers = self.inbound_handlers.clone();
-        task::spawn(async move {
-            if let Err(e) = inbound_handlers.handle_transaction(transaction, source_peer_id).await {
-                error!(
-                    target: LOG_TARGET,
-                    "Failed to handle incoming transaction message: {:?}", e
-                );
+        if let Err(e) = inbound_handlers.handle_transaction(transaction, source_peer_id).await {
+            error!(
+                target: LOG_TARGET,
+                "Failed to handle incoming transaction message: {:?}", e
+            );
+            // NOTE: We assume that all errors are due to a "bad" transaction.
```

Comment:
I feel like this might be a bit broad...
But we can use this for now, just we have to monitor it

Reply:
Yeah could be a problem - let's see.